C-18-01-Capstone.Main/DataAccessLayer/DataAccess.cs

```diff
@@ -1,26 +1,45 @@
 ﻿using C_18_01_Capstone.Main.DataContext;
-
+using System.Collections.Generic;
+using System.Linq;
 
 namespace C_18_01_Capstone.Main.DataAccessLayer
 {
-   public class DataAccess<T> : IDataAccess<T> where T : class
-   {
-      public void AddEntity(T entity)
-      {
-         using (var context = new SocialNetworkContext())
-         {
-            context.Set<T>().Add(entity);
-            context.SaveChanges();
-         }
-      }
+    public class DataAccess<T> : IDataAccess<T> where T : class
+    {
+        public void AddEntity(T entity)
+        {
+            using (var context = new SocialNetworkContext())
+            {
+                context.Set<T>().Add(entity);
+                context.SaveChanges();
+            }
+        }
+
+        public void AddRange(IEnumerable<T> entities)
+        {
+            foreach (var item in entities)
+            {
+                AddEntity(item);
```

Comment:
SaveChanges will be called each time, when AddEntity called. This is very inefficient, refactor this.